File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L2/07/Kasus7L2.kt
package IR

import java.util.Scanner

fun main() {
    // minta input
    val inp = Scanner(System.`in`)
    println("Enter a 4 by 4 matrix row by row: ")
    val matrix = Array(4) { DoubleArray(4) }
    // input the matrix
    for (i in 0 until 4) {
        for (j in 0 until 4) {
            matrix[i][j] = inp.nextDouble()
        }
    }
    println("Sum of the elements in the major diagonal is ${jumlahDiagonalKanan(matrix)}")
}

fun jumlahDiagonalKanan(matrix: Array<DoubleArray>): Double {
    var sum = 0.0
    for (i in 0 until matrix.size) {
        sum += matrix[i][i]
    }
    return sum
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L2/07/Kasus7L2.ts
import { Scanner } from 'readline-sync';

function jumlahDiagonalKanan(matrix: number[][]): number {
  let sum = 0;
  for (let i = 0; i < matrix.length; i++) {
    sum += matrix[i][i];
  }
  return sum;
}

function main(): void {
  const inp = new Scanner();
  console.log('Enter a 4 by 4 matrix row by row:');
  const matrix: number[][] = [];
  for (let i = 0; i < 4; i++) {
    const row: number[] = [];
    for (let j = 0; j < 4; j++) {
      row.push(inp.nextDouble());
    }
    matrix.push(row);
  }
  console.log(`Sum of the elements in the major diagonal is ${jumlahDiagonalKanan(matrix)}`);
}

main();

